arch/arc/include/swap_macros.h

```diff
@@ -247,7 +261,7 @@ extern "C" {
  * _kernel.current. r3 is a scratch reg.
  */
 .macro _load_stack_check_regs
-#ifdef CONFIG_ARC_HAS_SECURE
+#if defined(CONFIG_ARC_HAS_SECURE) && defined(CONFIG_ARC_SECURE_FIRMWARE)
```

Comment:
Missing case for HAS_SECURE and NORMAL_FIRMWARE?

Reply:
fixed